wallet/walletrpc.c

```diff
@@ -82,17 +113,50 @@ static u8 *scriptpubkey_from_address(const tal_t *cxt, bool *testnet,
 {
 	struct bitcoin_address p2pkh_destination;
 	struct ripemd160 p2sh_destination;
+	int witness_version;
+	uint8_t witness_program[40];
+	size_t witness_program_len;
+	bool witness_ok;
 	u8 *script = NULL;
 
+	char *addrz;
+	bool my_testnet;
+
 	if (bitcoin_from_base58(testnet, &p2pkh_destination,
 				addr, addrlen)) {
 		script = scriptpubkey_p2pkh(cxt, &p2pkh_destination);
 	} else if (p2sh_from_base58(testnet, &p2sh_destination,
 				    addr, addrlen)) {
 		script = scriptpubkey_p2sh_hash(cxt, &p2sh_destination);
 	}
-	/* TODO Insert other supported addresses here. */
+	/* Insert other parsers that accept pointer+len here. */
+
+	if (script) return script;
+
+	/* Generate null-terminated address. */
+	addrz = tal_arr(cxt, char, addrlen + 1);
+	memcpy(addrz, addr, addrlen);
+	addrz[addrlen] = '\0';
```

Comment:
Minor note:
    addrz = tal_dup_arr(ctx, char, addr, addrlen, 1);

Will do this in one line.